packages/react-html/README.md

```diff
@@ -12,112 +12,267 @@ $ yarn add @shopify/react-html
+
+function Serialize({data}: {data(): string}): null;
+function WithSerialized({
+  children,
+}: {
+  children(data: string | undefined): React.ReactNode;
+}): React.ReactNode;
 ```
 
-### Basic props
+The general pattern for using these components is to render the `WithSerialized` component as the top-most child. Within the render prop, construct whatever stateful store/ manager you need, using the data that was retrieved in cases where the serialization was found (on the browser, or on subsequent server renders). Finally, render the UI that depends on that stateful part, and a `Serialize` component that extracts the part that you you need to communicate between server and client.
 
-Most simple applications will only need these basic properties.
+Here is a complete example, using `@shopify/react-i18n`’s support for async translations as the data that needs to be serialized:
 
-**children**
-The children to be rendered inside the `#app` div.
+```tsx
+import {createSerializer} from '@shopify/react-html';
+import {Provider, Manager} from '@shopify/react-i18n';
 
-**styles**
-Descriptors for any style tags you want to include in the HEAD of the document.
+interface Props {
+  locale: string;
+  children: React.ReactNode;
+}
 
-**scripts**
-Descriptors for any script tags you want to include in your document. All scripts passed to this property will be deferred by appending them to the end of the document. We encourage this as a default, although you may use `blockingScripts` for any scripts that must be included in the HEAD of the document.
+const {Serialize, WithSerialized} = createSerializer<
+  ReturnType<Manager['extract']>
+>('i18n');
+
+export default function I18n({locale, children}: Props) {
+  return (
+    <WithSerialized>
+      {translations => {
+        const manager = new Manager({locale}, translations);
+
+        return (
+          <>
+            <Provider manager={manager}>{children}</Provider>
+            <Serialize
+              data={() => ({
+                locale: manager.details.locale,
+                translations: manager.extract(),
+              })}
+            />
+          </>
+        );
+      }}
+    </WithSerialized>
+  );
+}
+```
 
-**blockingScripts**
-Descriptors for any script tags you want to include in the HEAD of the document. These will block HTML parsing until they are evaluated, so use them carefully.
+#### Utility functions
 
-**hideForInitialLoad**
-Sets the body contents to be hidden for the initial render. Use this when injecting stylesheets dynamically in development in order to prevent a flash of unstyled content.
+##### `showPage()`
 
-### Serializers
+This function encapsulates the logic for showing the page in development, where it is hidden on the initial render by default. You must call this function from your client entry point, usually right after hydrating your React app. It returns a promise that resolves after the document is guaranteed to be visible.
```

Comment:
If you *must* call this as part of the client entry point it doesn't really sound like a "Utility" (which makes me think "helper" and "optional").

Placing this example of how to set up your client entry point near the example of how to set up your server entry would be good to make it clear there are multiple steps.

I'd put those preliminary requirements at the start of the usage section, before talking about how to configure a component.

## Usage
### Configuring your Server Entry Point
### Configuring your Client Entry Pont
### Using createSerializer() within a component (but with a better name)